### ISSUE TITLE
Track VDK SDK usage

### ISSUE THREAD

**Issue by antoniivanov** [vmware/versatile-data-kit#2849]
**What is the feature request? What problem does it solve?**

We want to track and measure the real usage of a  VDK SDK (quickstart-vdk and different major plugins) in reliable way excluding bots and automation (like our CICD) 

Currently we have some insights here https://github.com/vmware/versatile-data-kit/pulse 
and stats in https://pypistats.org like https://pypistats.org/packages/quickstart-vdk 

There is also this report in mode https://app.mode.com/vmware_inc/reports/179589458c9a

**Suggested solution**

Those are just ideas. And they are not that good. So hopeful for more

* Download Stats are available in BitQuery table ([click here](https://cloud.google.com/blog/topics/developers-practitioners/analyzing-python-package-downloads-bigquery) ) and could be used . If we can figure out a way to isolate real downloads from fake.

* During vdk create or other operations (appropriate plugin hook) we can download some resources (like getting started) - which can be hidden behind bit.ly proxy - bit.ly has some statistics for how many people a link has been accessed.  More advanced telemetry service could be used. 
This approach should really be opt out if if we are not collecting anything really , just pings.  That is to allow users to opt out. But also our CICD would opt out as well. 

**Acceptance criteria**
#1: We have agreed approach for tracking which users are using VDK. 
#2: Provide the infrastructure in VDK that follows the approach in #1
#3: Have some statistics.

**Comment by yonitoo** [comment]
The following ideas were tested:

- Querying the Google BigQuery tables
With the power of the BigQuery API, I was able to query the publicly available PyPI download statistics dataset and gather more and useful statistics than the PyPIstats provides. After some tests, the results show us that we can say confidently (with a possible insignificant statistical error) that we are able to isolate the real downloads from the CI/CD ones. The numbers would be soon publicly visible.
- Using bit.ly:
The main point of not going with it is that it does not provide very detailed information about the usage (clicks, geolocation). 
Out of the few tests I did, I couldn't isolate effectively the real from the CI/CD actions. Might be investigated further in the future.
- Using VMware SuperCollider:
Internal legal approval is needed in order to use SuperCollider for our purposes so we are leaving this approach on hold for now.
- Using Google Analytics (GA4):
It turned out there is a significant challenge in safely implementing Google Analytics (GA4) in the SDK context. The concern is that the GA4 Measurement Protocol requires an API Secret to send requests, and this API Secret is sensitive information. Exposing it to the SDK users would not be secure. One way to mitigate this issue would be to provide an intermediary endpoint where users send their usage data, and then this endpoint would forward the data to GA4. However, the VDK SDK doesn’t operate with a server to expose such an endpoint. Therefore, we will refrain from using Google Analytics for tracking in our SDK at this point. 

Summary: At this point, we can take advantage of the Google BigQuery API in order to estimate the number of real downloads of each VDK-related package.
